### PR TITLE
Add the STL_UNLOAD_LOG model

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
The plan is to model all the STL tables here, however, currently I only need to access this table.

**Comment by jklukas** [comment]
I guess these system tables aren't likely to change often, but modeling them seems like a potentially large maintenance burden. If we want to model all these tables, I'd advocate for building a script that can scrape the Redshift docs on the web and spit out the relevant code to create these models. It would give us a repeatable way to see if anything in the docs has changed.

This comment shouldn't prevent us going ahead with this PR, though.

File: redshift_sqlalchemy/models.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    This table is visible to all users.
+    """
+
+    __tablename__ = 'STL_UNLOAD_LOG'
+
+    user_id = sa.Column('userid', sa.Integer())
+    """ID of the user who generated the entry."""
+
+    query_id = sa.Column('query', sa.Integer(), primary_key=True)
+    """ID for the transaction."""
```
Comment (jklukas) [inline redshift_sqlalchemy/models.py]: I'm guessing it's a hard requirement that a declarative base class has a primary key?

The combination of `query_id` and `path` seems like a reasonable guarantee of uniqueness, but I'm generally uncomfortable with the idea of having to impose a primary key where none exists in the actual schema.

Comment (jklukas) [inline redshift_sqlalchemy/models.py]: I'd almost be inclined to define all columns as part of the primary key. That would at least be a consistent pattern that avoids us having to make an arbitrary choice for each new table we model.

Comment (graingert) [inline redshift_sqlalchemy/models.py]: The primary key is implicitly stated in the docs here

File: redshift_sqlalchemy/models.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    """Start time for the transaction."""
+
+    end_time = sa.Column(sa.DateTime())
+    """End time for the transaction."""
+
+    line_count = sa.Column(sa.BigInteger())
+    """Number of lines (rows) unloaded to the file."""
+
+    transfer_size = sa.Column(sa.BigInteger())
+    """Number of bytes transferred."""
```
Comment (jklukas) [inline redshift_sqlalchemy/models.py]: I've never seen docstrings for variables before. I know this isn't directly supported in python, but is this something that Sphinx is able to pick up somehow? If so, my mind will be blown.

**Comment by jklukas** [comment]
This looks good to me, but I have no experience with declarative bases.

File: redshift_sqlalchemy/models.py (hunk 1 above)
Comment (jklukas) [inline redshift_sqlalchemy/models.py]: As long as `query_id` is guaranteed to be unique, that seems reasonable.

File: redshift_sqlalchemy/models.py (hunk 2 above)
Comment (graingert) [inline redshift_sqlalchemy/models.py]: Yup consider your mind blown, Sphinx generates nice looking docs for these

https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring

> String literals occurring elsewhere in Python code may also act as documentation. They are not recognized by the Python bytecode compiler and are not accessible as runtime object attributes (i.e. not assigned to __doc__ ), but two types of extra docstrings may be extracted by software tools:

>    1. String literals occurring immediately after a simple assignment at the top level of a module, class, or __init__ method are called "attribute docstrings".
>    2. String literals occurring immediately after another docstring are called "additional docstrings".

File: redshift_sqlalchemy/models.py (hunk 1 above)
Comment (graingert) [inline redshift_sqlalchemy/models.py]: it's the primary key of the `STL_QUERY` table

**Comment by graingert** [comment]
@jklukas you can access the real table object at `redshift_sqlalchemy.models.STLUploadLog.__table__`

**Comment by graingert** [comment]
I'd love to automate this, but there's going to be some issues eg, finding the pirmary key and declaring relaitionships.

File: redshift_sqlalchemy/models.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+import sqlalchemy as sa
+from sqlalchemy.ext import declarative
+
+
+Base = declarative.declarative_base()
+
+
+class STLUnloadLog(Base):
```
Comment (jklukas) [inline redshift_sqlalchemy/models.py]: I'd strongly prefer this be named `StlUnloadLog`. Capitalizing just the first letter of abbreviations is the most consistent way to render names as CamelCase.

Comment (graingert) [inline redshift_sqlalchemy/models.py]: https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles
> Note: When using abbreviations in CapWords, capitalize all the letters of the abbreviation. Thus HTTPServerError is better than HttpServerError.

Comment (jklukas) [inline redshift_sqlalchemy/models.py]: Nooooooooooo! I feel betrayed by PEP 8.

I always considered the capitalization of SimpleHTTPServer to be a mistake, but apparently it was consistent all along. TIL.

Comment (graingert) [inline redshift_sqlalchemy/models.py]: I want to tear my eyes out when I see "Http"

**Comment by graingert** [comment]
Woops, forgot a changelog!

File: redshift_sqlalchemy/models.py (hunk 3 above)
Comment (jklukas) [inline redshift_sqlalchemy/models.py]: I guess it's all what you're used to.

**Comment by jklukas** [comment]
I think this is ready once it's added to the changelog.

**Comment by jklukas** [comment]
This will also need a rebase on master and a directory change for the models file to match the new package name.

**Comment by graingert** [comment]
I don't need this model anymore

**Pull request closed without merging**